chain/client/src/client.rs

```diff
@@ -1027,11 +1039,11 @@ impl Client {
         &mut self,
         shard_uid: ShardUId,
         prev_block: &Block,
+        prev_chunk_opt: Option<&ShardChunk>,
```

Comment:
I remember that some tests used to fail because of `prev_chunk` being `None`, something around genesis and missing chunks, but I tested it again and nothing fails. I guess it was something specific to the previous version of the code. I'll remove the `Option`